yarn-project/end-to-end/src/setup.ts

```diff
@@ -0,0 +1,38 @@
+import { AztecNodeConfig, AztecNodeService, getConfigEnvVars } from '@aztec/aztec-node';
+import { DebugLogger, createDebugLogger } from '@aztec/foundation/log';
+
+import { DeployL1Contracts, deployL1Contracts } from '@aztec/ethereum';
+import { mnemonicToAccount } from 'viem/accounts';
+import { createAztecRpcServer } from './create_aztec_rpc_client.js';
+import { MNEMONIC, localAnvil } from './fixtures.js';
+import { AztecAddress, AztecRPCServer } from '@aztec/aztec.js';
+
+/**
+ * Sets up the environment for the end-to-end tests.
+ * @param numberOfAccounts - The number of new accounts to be created once the RPC server is initiated.
+ * @returns A tuple containing the Aztec Node service, the Aztec RPC server and a logger.
+ */
+export async function setup(
+  numberOfAccounts = 1,
+): Promise<[AztecNodeService, AztecRPCServer, DeployL1Contracts, AztecAddress[], AztecNodeConfig, DebugLogger]> {
+  const config = getConfigEnvVars();
+
+  const describeBlockName = expect.getState().currentTestName?.split(' ')[0];
+
+  const logger = createDebugLogger('aztec:' + describeBlockName);
+
+  const hdAccount = mnemonicToAccount(MNEMONIC);
+  const privKey = hdAccount.getHdKey().privateKey;
+  const deployL1ContractsValues = await deployL1Contracts(config.rpcUrl, hdAccount, localAnvil, logger);
+
+  config.publisherPrivateKey = Buffer.from(privKey!);
+  config.rollupContract = deployL1ContractsValues.rollupAddress;
+  config.unverifiedDataEmitterContract = deployL1ContractsValues.unverifiedDataEmitterAddress;
+  config.inboxContract = deployL1ContractsValues.inboxAddress;
+
+  const node = await AztecNodeService.createAndSync(config);
+  const aztecRpcServer = await createAztecRpcServer(numberOfAccounts, node);
+  const accounts = await aztecRpcServer.getAccounts();
+
+  return [node, aztecRpcServer, deployL1ContractsValues, accounts, config, logger];
```

Comment:
Think it would be nicer and easier to decompose if this was returned as an object instead of array; e.g. you would have:
```
const { node, logger } = setup()
```
instead of
```
const [node, , , , , logger] = setup()
```

Reply:
Addressed in [30b2d81](https://github.com/AztecProtocol/aztec-packages/pull/708/commits/30b2d81c8ffd288526038a191468911327244b47)